From: oscar <oscar@wanadoo.es>
Date: Thu, 10 Oct 2019 16:33:54 +0200
Subject: [PATCH] lldb: fix CMake when LLDB_DISABLE_PYTHON if ON and epydoc is
 present

https://bugs.llvm.org/show_bug.cgi?id=43637

---
 docs/CMakeLists.txt | 6 ++++--
 1 file changed, 4 insertions(+), 2 deletions(-)

diff --git a/docs/CMakeLists.txt b/docs/CMakeLists.txt
index bfdd50badf6..a15f0e2db38 100644
--- a/docs/CMakeLists.txt
+++ b/docs/CMakeLists.txt
@@ -18,7 +18,11 @@ endif()
 find_program(EPYDOC_EXECUTABLE NAMES epydoc epydoc.py)
 if(EPYDOC_EXECUTABLE)
   message(STATUS "Found epydoc - ${EPYDOC_EXECUTABLE}")
+else()
+  message(STATUS "Could NOT find epydoc")
+endif()
 
+if(NOT LLDB_DISABLE_PYTHON AND EPYDOC_EXECUTABLE)
   find_program(DOT_EXECUTABLE dot)
   if(DOT_EXECUTABLE)
     set(EPYDOC_OPTIONS ${EPYDOC_OPTIONS} --graph all --dotpath ${DOT_EXECUTABLE})
@@ -49,8 +53,6 @@ if(EPYDOC_EXECUTABLE)
     COMMENT "Generating LLDB Python API reference with epydoc" VERBATIM
   )
   add_dependencies(lldb-python-doc swig_wrapper lldb-python-doc-package)
-else()
-  message(STATUS "Could NOT find epydoc")
 endif()
 
 if (LLVM_ENABLE_SPHINX)
-- 
2.23.0

